java.lang.IllegalStateException: Failed to execute CommandLineRunner : for default and dev profiles

```
java.lang.IllegalStateException: Failed to execute CommandLineRunner

	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:798) ~[spring-boot-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:779) ~[spring-boot-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.boot.SpringApplication.run(SpringApplication.java:322) ~[spring-boot-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1237) ~[spring-boot-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) ~[spring-boot-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at eu.yadev.storeme.StoreMeApplication.main(StoreMeApplication.java:20) ~[classes!/:0.0.1-SNAPSHOT]

	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]

	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]

	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]

	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]

	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[storeme.jar:0.0.1-SNAPSHOT]

	at org.springframework.boot.loader.Launcher.launch(Launcher.java:107) ~[storeme.jar:0.0.1-SNAPSHOT]

	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[storeme.jar:0.0.1-SNAPSHOT]

	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88) ~[storeme.jar:0.0.1-SNAPSHOT]

Caused by: org.springframework.dao.InvalidDataAccessApiUsageException: Target object must not be null; nested exception is java.lang.IllegalArgumentException: Target object must not be null

	at org.springframework.orm.jpa.EntityManagerFactoryUtils.convertJpaAccessExceptionIfPossible(EntityManagerFactoryUtils.java:374) ~[spring-orm-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:257) ~[spring-orm-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:528) ~[spring-orm-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61) ~[spring-tx-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:242) ~[spring-tx-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:154) ~[spring-tx-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:178) ~[spring-data-jpa-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:95) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at com.sun.proxy.$Proxy127.save(Unknown Source) ~[na:na]

	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]

	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]

	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]

	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]

	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:205) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at com.sun.proxy.$Proxy89.save(Unknown Source) ~[na:na]

	at eu.yadev.storeme.StoreMeApplication.run(StoreMeApplication.java:33) ~[classes!/:0.0.1-SNAPSHOT]

	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:795) ~[spring-boot-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	... 13 common frames omitted

Caused by: java.lang.IllegalArgumentException: Target object must not be null

	at org.springframework.util.Assert.notNull(Assert.java:201) ~[spring-core-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.beans.AbstractNestablePropertyAccessor.setWrappedInstance(AbstractNestablePropertyAccessor.java:195) ~[spring-beans-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.beans.BeanWrapperImpl.setWrappedInstance(BeanWrapperImpl.java:153) ~[spring-beans-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.beans.AbstractNestablePropertyAccessor.setWrappedInstance(AbstractNestablePropertyAccessor.java:183) ~[spring-beans-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.beans.AbstractNestablePropertyAccessor.<init>(AbstractNestablePropertyAccessor.java:122) ~[spring-beans-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.beans.BeanWrapperImpl.<init>(BeanWrapperImpl.java:103) ~[spring-beans-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.data.util.DirectFieldAccessFallbackBeanWrapper.<init>(DirectFieldAccessFallbackBeanWrapper.java:36) ~[spring-data-commons-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.data.jpa.repository.support.JpaMetamodelEntityInformation.getId(JpaMetamodelEntityInformation.java:159) ~[spring-data-jpa-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.data.repository.core.support.AbstractEntityInformation.isNew(AbstractEntityInformation.java:42) ~[spring-data-commons-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.data.jpa.repository.support.JpaMetamodelEntityInformation.isNew(JpaMetamodelEntityInformation.java:246) ~[spring-data-jpa-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.data.jpa.repository.support.SimpleJpaRepository.save(SimpleJpaRepository.java:553) ~[spring-data-jpa-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]

	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]

	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]

	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]

	at org.springframework.data.repository.core.support.ImplementationInvocationMetadata.invoke(ImplementationInvocationMetadata.java:72) ~[spring-data-commons-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.data.repository.core.support.RepositoryComposition$RepositoryFragments.invoke(RepositoryComposition.java:382) ~[spring-data-commons-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.data.repository.core.support.RepositoryComposition.invoke(RepositoryComposition.java:205) ~[spring-data-commons-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.data.repository.core.support.RepositoryFactorySupport$ImplementationMethodExecutionInterceptor.invoke(RepositoryFactorySupport.java:549) ~[spring-data-commons-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.doInvoke(QueryExecutorMethodInterceptor.java:155) ~[spring-data-commons-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.data.repository.core.support.QueryExecutorMethodInterceptor.invoke(QueryExecutorMethodInterceptor.java:130) ~[spring-data-commons-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.data.projection.DefaultMethodInvokingMethodInterceptor.invoke(DefaultMethodInvokingMethodInterceptor.java:80) ~[spring-data-commons-2.3.4.RELEASE.jar!/:2.3.4.RELEASE]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:367) ~[spring-tx-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:118) ~[spring-tx-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:139) ~[spring-tx-5.2.9.RELEASE.jar!/:5.2.9.RELEASE]

	... 29 common frames omitted

```

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -34,7 +34,8 @@ pipeline {
                        namespace: 'dev-storeme',
                        workloadName: 'storeme-api',
                        imageRepo: env.DOCKER_REPO,
-                       imageTag: env.BRANCH_NAME
+                       imageTag: env.BRANCH_NAME,
+                       containerName: 'api'
                     )
                 }
             }

--- a/src/main/java/eu/yadev/storeme/StoreMeApplication.java
+++ b/src/main/java/eu/yadev/storeme/StoreMeApplication.java
@@ -7,6 +7,8 @@ import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
+import java.util.Optional;
+
 @SpringBootApplication
 public class StoreMeApplication implements CommandLineRunner {
 
@@ -22,16 +24,22 @@ public class StoreMeApplication implements CommandLineRunner {
 
 	@Override
 	public void run(String... args) {
-		Record storyOne = Record.create().setTitle("Story One").setBody("Lorem ipsum dolor sit amet, consectetur adipiscing elit.");
-		Record storyTwo = Record.create().setTitle("Story Two").setBody("Donec tempor tempor ex.");
-		Record storyThree = Record.create().setTitle("Story Three").setBody(" Quisque laoreet volutpat felis in vulputate.");
+		Record recordOne = Record.create().setTitle("Story One").setBody("Lorem ipsum dolor sit amet, consectetur adipiscing elit.");
+		Record recordTwo = Record.create().setTitle("Story Two").setBody("Donec tempor tempor ex.");
+		Record recordThree = Record.create().setTitle("Story Three").setBody(" Quisque laoreet volutpat felis in vulputate.");
 
-		storyOne = recordDao.findByTitle(storyOne.getTitle());
-		storyTwo = recordDao.findByTitle(storyTwo.getTitle());
-		storyThree = recordDao.findByTitle(storyThree.getTitle());
+		Optional<Record> storyOne = recordDao.findSingleByTitle(recordOne.getTitle());
+		Optional<Record> storyTwo = recordDao.findSingleByTitle(recordTwo.getTitle());
+		Optional<Record> storyThree = recordDao.findSingleByTitle(recordThree.getTitle());
 
-		recordDao.save(storyOne);
-		recordDao.save(storyTwo);
-		recordDao.save(storyThree);
+		if (storyOne.isEmpty()) {
+			recordDao.save(recordOne);
+		}
+		if (storyTwo.isEmpty()) {
+			recordDao.save(recordTwo);
+		}
+		if (storyThree.isEmpty()) {
+			recordDao.save(recordThree);
+		}
 	}
 }

--- a/src/main/java/eu/yadev/storeme/models/dao/RecordDao.java
+++ b/src/main/java/eu/yadev/storeme/models/dao/RecordDao.java
@@ -2,9 +2,9 @@ package eu.yadev.storeme.models.dao;
 
 import eu.yadev.storeme.models.Record;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.stereotype.Repository;
 
-@Repository
+import java.util.Optional;
+
 public interface RecordDao extends JpaRepository<Record, Long> {
-	Record findByTitle(String title);
+	Optional<Record> findSingleByTitle(String title);
 }

